web3/providers/eth_tester/main.py

```diff
@@ -106,11 +106,11 @@ def make_request(self, method: RPCEndpoint, params: Any) -> RPCResponse:
                 "error": "RPC Endpoint has not been implemented: {0}".format(method),
             })
         except TransactionFailed as e:
-            if type(e.args[0]) == str:
-                reason = e.args[0]
-            else:
+            try:
                 reason = decode_single('(string)', e.args[0].args[0][4:])[0]
-            raise SolidityError(f'execution reverted: {reason}')
+            except (InsufficientDataBytes, AttributeError):
+                reason = e.args[0]
+            raise TransactionFailed(f'execution reverted: {reason}')
```

Comment:
The `TransactionError` vs. `TransactionFailed` distinction is a little fuzzy. If it happens in eth-tester, its the latter?

Reply:
Yeah, agreed. `TransactionError` comes from web3, whereas `TransactionFailed` is from eth-tester. I'm not opposed to making them both `TransactionFailed`, but thought it would be confusing to have them both be the same thing.

Reply:
My first instinct is to pick one generic error for both libs if they're used in the same context. (Might be wise to check what else can trigger this exception in eth-tester first.) Adopting eth-tester's `TransactionFailed` in web3 seems reasonable to me on the surface, though.

Reply:
Yeah, I'm good with that. It looks to me like eth-tester is using it in the same context. It looks like it's used for `revert` and `invalid` opcodes on `call` and `estimateGas` AFAICT.

Reply:
Why not `AssertionError` since it stems from a `REVERT` opcode, which usually in context is from an `assert`/`require` statement triggering (for any EVM language)?

Should add a test using a v0.8.0 Solidity contract or Vyper contract with checked arithmetic, since that's not a user revert but the language implementing a logical revert (that doesn't stem from an EVM exception)

Reply:
After polling the team, we landed on a `RevertTransaction` since that's the opcode. I added an explicit comment to #1812 to check for the logical revert. Thanks @fubuloubu!